src/panels/config/integrations/integration-panels/zha/zha-network-visualization-page.ts

```diff
@@ -0,0 +1,237 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+  PropertyValues,
+  query,
+} from "lit-element";
+
+import { navigate } from "../../../../../common/navigate";
+import { fetchDevices, ZHADevice } from "../../../../../data/zha";
+import "../../../../../layouts/hass-error-screen";
+import "../../../../../layouts/hass-subpage";
+import type { HomeAssistant } from "../../../../../types";
+import { Network, Edge, Node, EdgeOptions } from "vis-network";
+import "vis-network/styles/vis-network.css";
+import "../../../ha-config-section";
+
+@customElement("zha-network-visualization-page")
+export class ZHANetworkVisualizationPage extends LitElement {
+  @property({ type: Object }) public hass!: HomeAssistant;
+
+  @property({ type: Boolean }) public narrow!: boolean;
+
+  @query("#visualization", true)
+  private _visualization!: HTMLElement;
+
+  @internalProperty()
+  private _firstUpdatedCalled = false;
+
+  @internalProperty()
+  private _devices: ZHADevice[] = [];
+
+  @internalProperty()
+  private _network!: Network;
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    if (this.hass && this._firstUpdatedCalled) {
+      this._fetchData();
```

Comment:
I think that you are now going to fetch data twice on first render? Or is connected done after firstUpdated ?

Reply:
removed